app/src/main/java/org/simple/clinic/newentry/PatientEntryEffectHandler.kt

```diff
@@ -104,26 +106,27 @@ object PatientEntryEffectHandler {
   private fun savePatientTransformer(
       patientRepository: PatientRepository,
       patientRegisteredCount: Preference<Int>,
-      ui: PatientEntryUi,
-      schedulersProvider: SchedulersProvider
+      scheduler: Scheduler
   ): ObservableTransformer<SavePatient, PatientEntryEvent> {
     return ObservableTransformer { savePatientEffects ->
       savePatientEffects
           .map { it.entry }
-          .subscribeOn(schedulersProvider.io())
-          .flatMapCompletable {
-            patientRepository
-                .saveOngoingEntry(it)
-                .observeOn(schedulersProvider.ui())
-                .andThen(Completable.fromAction {
-                  patientRegisteredCount.set(patientRegisteredCount.get().plus(1))
-                  ui.openMedicalHistoryEntryScreen() // TODO(rj): 2019-10-04 Split this logic and dispatch a patient saved event + threading
-                })
-          }
-          .toObservable()
+          .subscribeOn(scheduler)
+          .flatMapSingle { savePatientEntry(it, patientRepository) }
+          .doOnNext { patientRegisteredCount.set(patientRegisteredCount.get().plus(1)) }
+          .map { PatientEntrySaved }
     }
   }
 
+  private fun savePatientEntry(
+      entry: OngoingNewPatientEntry,
+      patientRepository: PatientRepository
+  ): Single<Boolean> {
+    return patientRepository
+        .saveOngoingEntry(entry)
+        .andThen(Single.just(true))
+  }
+
```

Comment:
Since we are not using the result of this anywhere, should we just change the return type to `Single<Any>` instead?

Reply:
We could do a `Single<Unit>`

Reply:
Works.